src/containers/Tenant/Query/QueriesHistory/QueriesHistory.tsx

```diff
@@ -23,8 +27,9 @@ interface QueriesHistoryProps {
 
 function QueriesHistory({changeUserInput}: QueriesHistoryProps) {
     const dispatch = useTypedDispatch();
+    const [filter, setFilter] = React.useState('');
 
-    const queriesHistory = useTypedSelector(selectQueriesHistory);
+    const queriesHistory = useTypedSelector(selectQueriesHistory(filter));
```

Comment:
What do you think if we operate with filter in store? This allows to preserve filter value when tabs are switching, and all staff with filtering would be done inside redux selectors (it would be no need to pass `filter` there).

Reply:
fixed